File: META_Fady/Metastone_AdminMerche/src/tn/edu/esprit/model/Commande.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.edu.esprit.model;
import java.util.Date;

/**
 *
 * @author FadyBelY
 */
public class Commande {
//attribs
    private int Id_commande;
    private String Type_paiement;
    private String Date_commande;
    private int Prix_total;
    private String Adresse;
    private String Date_livraison;
    private int Id_utilisateur;
    private int Id_produit;
    private int Quantite_acheter;
//CTOR
    //ctor with ID
    public Commande(int Id_commande, String Type_paiement, String Date_commande, int Prix_total, String Adresse, String Date_livraison, int Id_utilisateur)
    {
        this.Id_commande = Id_commande;
        this.Type_paiement = Type_paiement;
        this.Date_commande = Date_commande;
        this.Prix_total = Prix_total;
        this.Adresse = Adresse;
        this.Date_livraison = Date_livraison;
        this.Id_utilisateur = Id_utilisateur;
    }
    //ctor w/o ID
    public Commande(String Type_paiement, String Date_commande, int Prix_total, String Adresse, String Date_livraison)
    {
        this.Type_paiement = Type_paiement;
        this.Date_commande = Date_commande;
        this.Prix_total = Prix_total;
        this.Adresse = Adresse;
        this.Date_livraison = Date_livraison;
    }
    //ctor default
    public Commande()
    {
        Type_paiement = "";
        Date_commande = "";
        Prix_total = 0;
        Adresse = "";
        Date_livraison = "";
    }
    
    //ctor with id_user
    public Commande(String Type_paiement, String Date_commande, int Prix_total, String Adresse, String Date_livraison, int Id_utilisateur,int Id_produit,int Quantite_acheter) {
        this.Type_paiement = Type_paiement;
        this.Date_commande = Date_commande;
        this.Prix_total = Prix_total;
        this.Adresse = Adresse;
        this.Date_livraison = Date_livraison;
        this.Id_utilisateur = Id_utilisateur;
        this.Id_produit = Id_produit;
        this.Quantite_acheter = Quantite_acheter;
    }

    public int getQuantite_acheter() {
        return Quantite_acheter;
    }

    public void setQuantite_acheter(int Quantite_acheter) {
        this.Quantite_acheter = Quantite_acheter;
    }
    public int getId_produit() {    
        return Id_produit;
    }

//GETTER & SETTER
    public void setId_produit(int Id_produit) {    
        this.Id_produit = Id_produit;
    }

    //GETTERS
    public int getId_commande() {
        return Id_commande;
    }

    public String getType_paiement() {
        return Type_paiement;
    }

    public String getDate_commande() {
        return Date_commande;
    }

    public int getPrix_total() {
        return Prix_total;
    }

    public String getAdresse() {
        return Adresse;
    }

    public String getDate_livraison() {
        return Date_livraison;
    }

    public int getId_utilisateur() {
        return Id_utilisateur;
    }
    
    //SETTERS
    public void setId_commande(int Id_commande) {
        this.Id_commande = Id_commande;
    }

    public void setType_paiement(String Type_paiement) {
        this.Type_paiement = Type_paiement;
    }

    public void setDate_commande(String Date_commande) {
        this.Date_commande = Date_commande;
    }

    public void setPrix_total(int Prix_total) {
        this.Prix_total = Prix_total;
    }

    public void setAdresse(String Adresse) {
        this.Adresse = Adresse;
    }

    public void setDate_livraison(String Date_livraison) {
        this.Date_livraison = Date_livraison;
    }

    public void setId_utilisateur(int Id_utilisateur) {
        this.Id_utilisateur = Id_utilisateur;
    }
    
//Affichage
    //to String
    @Override
    public String toString() {
        return "Commande{" + "Id_commande=" + Id_commande + ", Type_paiement=" + Type_paiement + ", Date_commande=" + Date_commande + ", Prix_total=" + Prix_total + ", Adresse=" + Adresse + ", Date_livraison=" + Date_livraison + ", Id_utilisateur=" + Id_utilisateur + '}';
    }
    
            
            
            
}

File: META_Moez/Metastone_web/src/Entity/Commande.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Commande
 *
 * @ORM\Table(name="commande", indexes={@ORM\Index(name="etrprod", columns={"Id_produit"}), @ORM\Index(name="etranger_cammande_user", columns={"Id_utilisateur"})})
 * @ORM\Entity
 */
class Commande
{
    /**
     * @var int
     *
     * @ORM\Column(name="Id_commande", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idCommande;

    /**
     * @var string
     *
     * @ORM\Column(name="Type_paiement", type="string", length=255, nullable=false)
     */
    private $typePaiement;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="Date_commande", type="date", nullable=false)
     */
    private $dateCommande;

    /**
     * @var float
     *
     * @ORM\Column(name="Prix_total", type="float", precision=10, scale=0, nullable=false)
     */
    private $prixTotal;

    /**
     * @var string
     *
     * @ORM\Column(name="Adresse", type="string", length=255, nullable=false)
     */
    private $adresse;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="Date_livraison", type="date", nullable=false)
     */
    private $dateLivraison;

    /**
     * @var int
     *
     * @ORM\Column(name="Quantite_acheter", type="integer", nullable=false)
     */
    private $quantiteAcheter;

    /**
     * @var \Merche
     *
     * @ORM\ManyToOne(targetEntity="Merche")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="Id_produit", referencedColumnName="Id_produit")
     * })
     */
    private $idProduit;

    /**
     * @var \Utilisateur
     *
     * @ORM\ManyToOne(targetEntity="Utilisateur")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="Id_utilisateur", referencedColumnName="ID_utilisateur")
     * })
     */
    private $idUtilisateur;

    public function getIdCommande(): ?int
    {
        return $this->idCommande;
    }

    public function getTypePaiement(): ?string
    {
        return $this->typePaiement;
    }

    public function setTypePaiement(string $typePaiement): self
    {
        $this->typePaiement = $typePaiement;

        return $this;
    }

    public function getDateCommande(): ?\DateTimeInterface
    {
        return $this->dateCommande;
    }

    public function setDateCommande(\DateTimeInterface $dateCommande): self
    {
        $this->dateCommande = $dateCommande;

        return $this;
    }

    public function getPrixTotal(): ?float
    {
        return $this->prixTotal;
    }

    public function setPrixTotal(float $prixTotal): self
    {
        $this->prixTotal = $prixTotal;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getDateLivraison(): ?\DateTimeInterface
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(\DateTimeInterface $dateLivraison): self
    {
        $this->dateLivraison = $dateLivraison;

        return $this;
    }

    public function getQuantiteAcheter(): ?int
    {
        return $this->quantiteAcheter;
    }

    public function setQuantiteAcheter(int $quantiteAcheter): self
    {
        $this->quantiteAcheter = $quantiteAcheter;

        return $this;
    }

    public function getIdProduit(): ?Merche
    {
        return $this->idProduit;
    }

    public function setIdProduit(?Merche $idProduit): self
    {
        $this->idProduit = $idProduit;

        return $this;
    }

    public function getIdUtilisateur(): ?Utilisateur
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(?Utilisateur $idUtilisateur): self
    {
        $this->idUtilisateur = $idUtilisateur;

        return $this;
    }


}
